### ISSUE TITLE
authConfig with API key is broken

### ISSUE THREAD

**Issue by vprivat-ads** [radiantearth/stac-browser#446]
I use this script to launch stac-browser with API key authentication:

```
sed -i 's/authConfig: null/authConfig: { "type": "header", "key": "x-api-key", "description": "test" }/g' config.js && npm install && \
	npm start -- --open \
	--catalogUrl="https://<my-domain>/catalog/" \
	--pathPrefix="/browser/" \
	--allowExternalAccess=false \
	--allowedDomains="<my-domain>" \
	--detectLocaleFromBrowser=true
```

It worked fine until a few days ago, now after providing my API key, the UI just "hangs" forever:

![image](https://github.com/radiantearth/stac-browser/assets/146961743/6b0f148e-13e3-48c7-8d90-08e45748527e)

**Comment by vprivat-ads** [comment]
last working commit is e725d3c89c2de8d0c8f58c95c32dc3527d1e5f07

after that it doesn't work anymore.

**Comment by m-mohr** [comment]
@vprivat-ads The PR seems completely unrelated to Auth. Does the browser have any error messages in the dev console?

Anyway, it will likely not be fixed with the current implementation. Instead https://github.com/radiantearth/stac-browser/pull/439 will be merged soon. Please check that the PR works for you and report any issues there.

**Comment by vprivat-ads** [comment]
I have the same behaviour between main branch and #439. When launching stac-browser, a first request is made without http header and results in a 403. Then, when I enter my API key, no further request is made, and the UI just hangs. I don't see any relevant error in dev tools.

Do you confirm the new syntax to use in #439 is `authConfig: { "type": "apiKey", "in": "header", "name": "x-api-key", "description": "test" }` ?

In my previous comment I posted the last working commit. The first one which fails is c09cbd7008a30026487b3423917b2dcbd3da0966 and is related to async actions. Maybe entering an API Key is not considered as a "config change" that triggers the required async action to perform the request to the catalog?

**Comment by m-mohr** [comment]
I see, that commit makes more sense. I'll investigate.

The new syntax looks good, although it should also internally convert correctly from the old syntax.

**Comment by m-mohr** [comment]
@vprivat-ads I think this should be fixed in #439. Could you please confirm whether it also works for you?

**Comment by vprivat-ads** [comment]
Hello Matthias,
Yes it works now in the #439 branch! Thanks a lot!
